indico/htdocs/js/indico/modules/contributions/common.js

```diff
@@ -105,48 +105,56 @@
         });
     }
 
+    function changeContribDisplayedStartDate(itemPicker) {
+        itemPicker.closest('tr').find('td.start-date').html($('<em>', {'text': $T.gettext('Not scheduled')}));
+    }
+
     function setupSessionPicker(createURL) {
         $('#contribution-list .session-item-picker').itempicker({
             filterPlaceholder: $T.gettext('Filter sessions'),
             containerClasses: 'session-item-container',
             footerElements: [{
-                title: $T.gettext('Add new session'),
-                onClick: function() {
+                title: $T.gettext('Assign new session'),
+                onClick: function(itemPicker) {
                     ajaxDialog({
                         title: $T.gettext('Add new session'),
                         url: createURL,
                         onClose: function(data) {
                             if (data) {
+                                var postData = {session_id: data.new_session_id};
                                 $('.session-item-picker').itempicker('refreshItemList', data.sessions);
+
+                                patchObject(itemPicker.data('href'), itemPicker.data('method'), postData).then(function(contribData) {
```

Comment:
You have the same call in two places (after session creation and on item selection). Why not do one of these things:

- move it to a function and call that function in both places
- add an option to selectItem whether to trigger `onSelect`

I think the second option would be nicer since it'd make the code that uses the widget smaller (after session creation would simply call `itemPicker.itempicker('selectItem', data.new_session_id, true);` and that's it)